sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -1672,13 +1672,18 @@ class Dataset[T] private[sql](
 
   /** @inheritdoc */
   override def toJSON: Dataset[String] = {
+    toJSON(Map.empty[String, String])
+  }
+
+  /** @inheritdoc */
+  override def toJSON(jsonOptions: Map[String, String]): Dataset[String] = {
```

Comment:
and .. actually we can easily work around via:

```scala
df.select(to_json(struct(col("*")), options)).as(StringEncoder)
```

Reply:
ah, we probably shouldn't add `Dataset.toJSON` at the first place...

Reply:
Yes, actually we can bypass it using this one properly, however, do you still think it is worth to extend the `toJSON` API? In my opinion, a lot of users will not figure out this workaround when trying to specify options for `toJSON` in a `DataSet`.